CMakeLists.txt

```
@@ -0,0 +1,57 @@
+cmake_minimum_required(VERSION 3.8)
+
+project(librw C CXX)
+
+if(PS2DEV AND EE)
+    include("${CMAKE_CURRENT_LIST_DIR}/cmake/ee.cmake")
+endif()
+
+if(WIN32)
+    set(LIBRW_PLATFORMS "NULL" "GL3" "D3D9")
+else()
+    set(LIBRW_PLATFORMS "NULL" "GL3" "PS2")
+endif()
+set(LIBRW_PLATFORM "NULL" CACHE STRING "Platform")
+set_property(CACHE LIBRW_PLATFORM PROPERTY STRINGS ${LIBRW_PLATFORMS})
+message(STATUS "LIBRW_PLATFORM = ${LIBRW_PLATFORM} (choices=${LIBRW_PLATFORMS})")
+set("LIBRW_PLATFORM_${LIBRW_PLATFORM}" ON)
+if(NOT LIBRW_PLATFORM IN_LIST LIBRW_PLATFORMS)
+    message(FATAL_ERROR "Illegal LIBRW_PLATFORM=${LIBRW_PLATFORM}")
+endif()
+
+if(LIBRW_PLATFORM_PS2)
+    enable_language(DSM)
+endif()
+
+option(LIBRW_TOOLS "Build librw tools" OFF)
+option(LIBRW_INSTALL "Install librw files" OFF)
```

I suggest enabling these option by default for higher coverage by default.
The tools cannot be built because the tools subdirectory does not contain a `CMakeLists.txt` file.
```suggestion
option(LIBRW_TOOLS "Build librw tools" ON)
option(LIBRW_INSTALL "Install librw files" ON)
```